cmd/agent/subcommands/status/command.go

```diff
@@ -155,27 +150,26 @@ func requestStatus(config config.Component, cliParams *cliParams) error {
 		v.Set("format", "text")
 	}
 
-	url := url.URL{
-		Scheme:   "https",
-		Host:     fmt.Sprintf("%v:%v", ipcAddress, config.GetInt("cmd_port")),
-		Path:     "/agent/status",
-		RawQuery: v.Encode(),
+	endpoint, err := apiutil.NewIPCEndpoint(config, "/agent/status")
+	if err != nil {
+		return err
 	}
+	endpoint = endpoint.WithValues(v)
 
-	r, err := makeRequest(url.String())
+	res, err := endpoint.DoGet()
```

Comment:
Seem this functions calls makes me think that we could pass the `url.vVlues` as part of the constructor of  `NewIPCEndpoint`. Also, the current implementation assumes that users `NewIPCEndpoint` would not want to configure the client, maybe to improve the constructor we could pass a http.CLient so caller can setup there own clients